src/test/java/shortestpath/PrimitiveIntListTests.java

```diff
@@ -67,4 +67,155 @@ public void tryRemove() {
         Assert.assertEquals(true, intList.remove(11));
         Assert.assertEquals(true, intList.isEmpty());
     }
+
+    @Test
+    public void defaultConstructor() {
+        PrimitiveIntList intList = new PrimitiveIntList();
+        Assert.assertEquals(0, intList.size());
+        Assert.assertTrue(intList.isEmpty());
+    }
+
+    @Test
+    public void capacityOnlyConstructor() {
+        PrimitiveIntList intList = new PrimitiveIntList(5);
+        // The list is not initialized so the size is 0
+        Assert.assertEquals(0, intList.size());
+        Assert.assertTrue(intList.isEmpty());
+        intList.add(42);
+        Assert.assertEquals(1, intList.size());
+        Assert.assertEquals(42, intList.get(0));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void negativeCapacityConstructor() {
+        new PrimitiveIntList(-1);
+    }
+
+        @Test
+    public void ZeroCapacityConstructor() {
```

Comment:
This should probably not have a capital first letter Z, so should instead be named `zeroCapacityConstructor`